### DISCUSSION TITLE
Unexpected OAuth redirect page response

### DISCUSSION THREAD

**Discussion by justin-souter** [etsy/open-api#1344] [Q&A] [answered]
My OAuth flow seems to be working fine, presenting the sign-in page and permissions request. When granted my redirect URI is called. However, instead of the documented access_token field in the redirect request, I'm getting "code" and "state" fields. The "code" field contains a 100 character string. I thought "code" was the access token, but it has no prefixed user id and no period separator (there's a couple of dashes in there though). Any advice?

**Comment by turboturtleD** [comment, marked as answer]
So the state value is for your own reference, and should match the state parameter from your request.  The code value I believe needs to be exchanged for a token.  Should all be in the documentation.

**Reply by justin-souter** [reply]
I looked right past that step several times in the "Quick Start". Thank you.
